types/aws-lambda/trigger/api-gateway-proxy.d.ts

```diff
@@ -66,6 +93,61 @@ export interface APIGatewayProxyResult {
     isBase64Encoded?: boolean;
 }
 
+/**
+ * Works with HTTP API integration Payload Format version 2.0
+ * @see - https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+ */
+export interface APIGatewayProxyEventV2 {
+    routeKey: string;
+    rawPath: string;
+    rawQueryString: string;
+    cookies: string[];
+    headers: { [name: string]: string };
+    queryStringParameters: { [name: string]: string };
+    requestContext: {
+        accountId: string;
+        apiId: string;
+        authorizer: {
+            jwt: {
+                claims: { [name: string]: string };
```

Comment:
Is it possible that this should be `string | number | boolean | string[]` ?

This is what Cognito claims looks like in my test.

Reply:
Yeah. Given that JWT specification allows string, array, boolean and integer as value of custom claims (which gets base64 encoded), it's only fitting that Lambda event supports them.

This PR has already been merged. Would you like to open a new PR?

Reply:
Created #45320

Reply:
Don't know why, but I don't see that top-level `jwt` inside authorizer. raised #50939